rust/cubesql/cubesql/src/compile/engine/information_schema/postgres/pg_collation.rs

```diff
@@ -0,0 +1,259 @@
+use std::sync::Arc;
+
+use async_trait::async_trait;
+
+use datafusion::{
+    arrow::{
+        array::{Array, ArrayRef, BooleanBuilder, Int32Builder, StringBuilder, UInt32Builder},
+        datatypes::{DataType, Field, Schema},
+        record_batch::RecordBatch,
+    },
+    datasource::{datasource::TableProviderFilterPushDown, TableProvider},
+    error::Result,
+    logical_plan::Expr,
+    physical_plan::{memory::MemoryExec, ExecutionPlan},
+};
+
+use crate::compile::engine::information_schema::postgres::PG_NAMESPACE_CATALOG_OID;
+
+struct PgCollation {
+    oid: u32,
+    collname: &'static str,
+    collnamespace: u32,
+    collowner: u32,
+    collprovider: String,
+    collisdeterministic: bool,
+    collencoding: i32,
+    collcollate: Option<String>,
+    collctype: Option<String>,
+    // Column `colliculocale` renamed to `colllocale` since PostgreSQL 17.
+    // Support both columns for backward-compatibility.
+    // Reference: https://pgpedia.info/p/pg_collation.html
+    colliculocale: Option<String>,
+    colllocale: Option<String>,
+    collicurules: Option<String>,
+    collversion: Option<String>,
+}
+
+struct PgCatalogCollationBuilder {
+    oid: UInt32Builder,
+    collname: StringBuilder,
+    collnamespace: UInt32Builder,
+    collowner: UInt32Builder,
+    collprovider: StringBuilder,
+    collisdeterministic: BooleanBuilder,
+    collencoding: Int32Builder,
+    colliculocale: StringBuilder,
+    collcollate: StringBuilder,
+    collctype: StringBuilder,
+    colllocale: StringBuilder,
+    collicurules: StringBuilder,
+    collversion: StringBuilder,
+}
+
+impl PgCatalogCollationBuilder {
+    fn new() -> Self {
+        let capacity = 5;
+        Self {
+            oid: UInt32Builder::new(capacity),
+            collname: StringBuilder::new(capacity),
+            collnamespace: UInt32Builder::new(capacity),
+            collowner: UInt32Builder::new(capacity),
+            collprovider: StringBuilder::new(capacity),
+            collisdeterministic: BooleanBuilder::new(capacity),
+            collencoding: Int32Builder::new(capacity),
+            colliculocale: StringBuilder::new(capacity),
+            collcollate: StringBuilder::new(capacity),
+            collctype: StringBuilder::new(capacity),
+            colllocale: StringBuilder::new(capacity),
+            collicurules: StringBuilder::new(capacity),
+            collversion: StringBuilder::new(capacity),
+        }
+    }
+    fn add_collation(&mut self, coll: &PgCollation) {
```

Comment:
Please run `cargo fmt` to fix formatting issues before CI kicks in and reports an error.